www/src/utils/pagination.ts

```diff
@@ -1,8 +1,8 @@
-import { SIDEBAR, Sidebar } from "../config";
+import { SIDEBAR, Sidebar, SidebarItem, SidebarItemLink } from "../config";
 
-export function paginate(lang: keyof Sidebar, path: string) {
-  const routes = Object.values(SIDEBAR[lang]).flat();
-  const index = routes.map((item) => item.link as string).indexOf(path);
+export function paginate(lang: keyof Sidebar, path: SidebarItemLink) {
+  const routes = Object.values(SIDEBAR[lang]).flat() as SidebarItem[];
```

Comment:
Why does using `Object.values` cast the array to `any` here? Maybe there is some way to fix the type at the source so we can avoid `as`?

Reply:
The actual type of `Object.values` (which you can skip to with go-to-definition) is:

```ts
{
    /**
     * Returns an array of values of the enumerable properties of an object
     * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
     */
    values<T>(o: { [s: string]: T } | ArrayLike<T>): T[];

    /**
     * Returns an array of values of the enumerable properties of an object
     * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
     */
    values(o: {}): any[];
}
```

In other words, `Object.values` has two type system understandings:

1. If the provided `o` has an index signature with values of type `T`, then the returned type is `T[]`
2. `any`

The type of `SIDEBAR[lang]` is:

```ts
{
    "Create T3 App"?: SidebarItem<"ar">[];
    Usage?: SidebarItem<"ar">[];
    Deployment?: SidebarItem<"ar">[];
} | {
    "Create T3 App"?: SidebarItem<"en">[];
    Usage?: SidebarItem<...>[];
    Deployment?: SidebarItem<...>[];
} | ... 4 more ... | {
    ...;
}
```

There's no index signature there. So we fall back to `any`. 😞 

https://github.com/microsoft/TypeScript/issues/38520 has some fun context around why things are the way they are.